code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -68,6 +67,72 @@
 			return -1
 	return 0
 
+
+
+//Organ
+
+/obj/item/organ/heart/nightmare
+	name = "heart of darkness"
+	desc = "An alien organ that twists and writhes when exposed to light."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "demon_heart-on"
+	color = "#1C1C1C"
+	var/respawn_progress = 0
+	var/obj/item/light_eater/blade
+
+
+
+/obj/item/organ/heart/nightmare/attack(mob/M, mob/living/carbon/user, obj/target)
+	if(M != user)
+		return ..()
+	user.visible_message("<span class='warning'>[user] raises [src] to their mouth and tears into it with their teeth!</span>", \
+						 "<span class='danger'>[src] feels unnaturally cold in your hands. You raise [src] your mouth and devour it!</span>")
+	playsound(user, 'sound/magic/demon_consume.ogg', 50, 1)
+
+
+	user.visible_message("<span class='warning'>Blood erupts from [user]'s arm as it reforms into a weapon!</span>", \
+						 "<span class='userdanger'>Icy blood pumps through your veins as your arm reforms itself!</span>")
+	user.drop_item()
+	src.Insert(user)
+
+/obj/item/organ/heart/nightmare/Insert(mob/living/carbon/M, special = 0)
+	..()
+	blade = new/obj/item/light_eater
+	M.put_in_hands(blade)
+	START_PROCESSING(SSobj, src)
+
+/obj/item/organ/heart/nightmare/Remove(mob/living/carbon/M, special = 0)
+	STOP_PROCESSING(SSobj, src)
+	respawn_progress = 0
+	if(blade)
+		QDEL_NULL(blade)
+		M.visible_message("<span class='warning'>\The [blade] disintegrates!</span>")
+	..()
+
+/obj/item/organ/heart/nightmare/Stop()
+	return 0
+
+/obj/item/organ/heart/nightmare/update_icon()
+	return //always beating visually
+
+/obj/item/organ/heart/nightmare/process()
+	if(!owner || owner.stat != DEAD)
+		respawn_progress = 0
+		return
+	var/turf/T = get_turf(owner)
+	if(istype(T))
+		var/light_amount = T.get_lumcount()
+		if(light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD)
+			respawn_progress++
+			playsound(owner,'sound/effects/singlebeat.ogg',40,1)
+	if(respawn_progress >= 40)
+		owner.revive(full_heal = TRUE)
+		owner.visible_message("<span class='warning'>[owner] staggers to their feet!</span>")
+		playsound(owner.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)

```

Comment:
playsound(owner, ...

Reply:
alternatively ```get_turf(owner)```

Reply:
playsound calls get_turf itself so that's redundant